src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThicknessConverter.cs

```diff
@@ -169,53 +162,66 @@ static internal string ToString(Thickness th, CultureInfo cultureInfo)
             // Initial capacity [64] is an estimate based on a sum of:
             // 48 = 4x double (twelve digits is generous for the range of values likely)
             //  8 = 4x Unit Type string (approx two characters)
-            //  4 = 4x separator characters
-            StringBuilder sb = new StringBuilder(64);
-
-            sb.Append(LengthConverter.ToString(th.Left, cultureInfo));
-            sb.Append(listSeparator);
-            sb.Append(LengthConverter.ToString(th.Top, cultureInfo));
-            sb.Append(listSeparator);
-            sb.Append(LengthConverter.ToString(th.Right, cultureInfo));
-            sb.Append(listSeparator);
-            sb.Append(LengthConverter.ToString(th.Bottom, cultureInfo));
-            return sb.ToString();
+            //  3 = 3x separator characters
+            //  1 = 1x scratch space for alignment
+
+            DefaultInterpolatedStringHandler handler = new(0, 7, cultureInfo, stackalloc char[64]);
+            FormatDoubleAsString(th.Left, ref handler);
+            handler.AppendFormatted(listSeparator);
+
+            FormatDoubleAsString(th.Top, ref handler);
+            handler.AppendFormatted(listSeparator);
+
+            FormatDoubleAsString(th.Right, ref handler);
+            handler.AppendFormatted(listSeparator);
+
+            FormatDoubleAsString(th.Bottom, ref handler);
+
+            return handler.ToStringAndClear();
         }
 
-        static internal Thickness FromString(string s, CultureInfo cultureInfo)
+        /// <summary> Holds the "Auto" string representation for <see cref="double.NaN"/> conversion. </summary>
+        private static ReadOnlySpan<char> NaNValue => ['A', 'u', 't', 'o'];
+
+        /// <summary> Format <see cref="double"/> into <see cref="string"/> using specified <see cref="CultureInfo"/>
+        /// in <paramref name="handler"/>. <br /> <br />
+        /// Special representation applies for <see cref="double.NaN"/> values, emitted as "Auto" string instead. </summary>
+        /// <param name="value">The value to format as string.</param>
+        /// <param name="handler">The handler specifying culture used for conversion.</param>
+        static internal void FormatDoubleAsString(double value, ref DefaultInterpolatedStringHandler handler)
         {
-            TokenizerHelper th = new TokenizerHelper(s, cultureInfo);
-            double[] lengths = new double[4];
+            if (double.IsNaN(value))
+                handler.AppendFormatted(NaNValue);
+            else
+                handler.AppendFormatted(value);
+        }
+
+        static internal unsafe Thickness FromString(string s, CultureInfo cultureInfo)
+        {
+            TokenizerHelper th = new(s, cultureInfo);
+            double* lengths = stackalloc double[4];
```

Comment:
You should use `Span<double>` here instead of `double*`:
```suggestion
            Span<double> lengths = stackalloc double[4];
```

Reply:
This was a consicous decision. I do not need the `Length`, the codegen is more efficient and 40 bytes shorter.

Reply:
Sorry for the late reply.

> This was a consicous decision. I do not need the `Length`, the codegen is more efficient and 60 bytes shorter.

IMO this isn't a good enough reason to use pointers. We should only use pointers when it is needed or when it is on the hottest of hot paths. The codegen is "more efficient and 60 bytes shorter" because it does less stuff. By using pointers we lose the safety of bound-checking just to save 60 bytes and maybe 1 or 2 ns. Using spans helps maintainability and helps avoiding mistakes.

Reply:
@h3xds1nz I too am of a similar opinion. Use of span would offer a better memory safety and readability.

Reply:
Alright, let the automatic bounds checks that should have been eliminated since "i" can never be bigger than the allocated length protect us and hope JIT will improve in the future here.